causalpy/experiments/interrupted_time_series.py

```diff
@@ -107,6 +110,33 @@ def __init__(
         )
         self.post_X = np.asarray(new_x)
         self.post_y = np.asarray(new_y)
+        # turn into xarray.DataArray's
+        self.pre_X = xr.DataArray(
```

Comment:
Seems like a pretty extensive API change for multiple different methods. Should we plan for removing patsy altogether? I do like formulas... but consistency is kind of important overall....

Reply:
I think keeping formulas as the main API is a good idea. See longer [reply](https://github.com/pymc-labs/CausalPy/pull/460#issuecomment-2865983360) below.